src/node_report.cc

```diff
@@ -0,0 +1,808 @@
+    if (static_cast<size_t>(i) < info.frames_count) {
+      PrintStackFrame(fp, isolate, stack->GetFrame(i), i, samples[i]);
+    } else {
+      PrintStackFrame(fp, isolate, stack->GetFrame(i), i, NULL);
+    }
+  }
+}
+
+/*******************************************************************************
+ * Function to print a JavaScript stack frame from a V8 StackFrame object
+ *
+ ******************************************************************************/
+static void PrintStackFrame(FILE* fp, Isolate* isolate, Local<StackFrame> frame, int i, void *pc) {
+  Nan::Utf8String fn_name_s(frame->GetFunctionName());
+  Nan::Utf8String script_name(frame->GetScriptName());
+  const int line_number = frame->GetLineNumber();
+  const int column = frame->GetColumn();
+
+  // First print the frame index and the instruction address
+#ifdef _WIN32
+  fprintf(fp, "%2d: [pc=0x%p] ", i, pc);
+#else
+  fprintf(fp, "%2d: [pc=%p] ", i, pc);
+#endif
+
+  // Now print the JavaScript function name and source information
+  if (frame->IsEval()) {
+    if (frame->GetScriptId() == Message::kNoScriptIdInfo) {
+      fprintf(fp, "at [eval]:%i:%i\n", line_number, column);
+    } else {
+      fprintf(fp, "at [eval] (%s:%i:%i)\n", *script_name, line_number, column);
+    }
+    return;
+  }
+
+  if (fn_name_s.length() == 0) {
+    fprintf(fp, "%s:%i:%i\n", *script_name, line_number, column);
+  } else {
+    if (frame->IsConstructor()) {
+      fprintf(fp, "%s [constructor] (%s:%i:%i)\n", *fn_name_s, *script_name, line_number, column);
+    } else {
+      fprintf(fp, "%s (%s:%i:%i)\n", *fn_name_s, *script_name, line_number, column);
+    }
+  }
+}
+
+
+#ifdef _WIN32
+/*******************************************************************************
+ * Function to print a native stack backtrace
+ *
+ ******************************************************************************/
+void PrintNativeStack(FILE* fp) {
+  void *frames[64];
+  fprintf(fp, "\n================================================================================");
+  fprintf(fp, "\n==== Native Stack Trace ========================================================\n\n");
+
+  HANDLE hProcess = GetCurrentProcess();
+  SymSetOptions(SYMOPT_LOAD_LINES | SYMOPT_UNDNAME | SYMOPT_DEFERRED_LOADS);
+  SymInitialize(hProcess, NULL, TRUE);
+
+  WORD numberOfFrames = CaptureStackBackTrace(2, 64, frames, NULL);
+  
+  // Walk the frames printing symbolic information if available
+  for (int i = 0; i < numberOfFrames; i++) {
+    DWORD64 dwOffset64 = 0;
+    DWORD64 dwAddress = (DWORD64)(frames[i]);
+    char buffer[sizeof(SYMBOL_INFO) + MAX_SYM_NAME * sizeof(TCHAR)];
+    PSYMBOL_INFO pSymbol = (PSYMBOL_INFO)buffer;
+    pSymbol->SizeOfStruct = sizeof(SYMBOL_INFO);
+    pSymbol->MaxNameLen = MAX_SYM_NAME;
+
+    if (SymFromAddr(hProcess, dwAddress, &dwOffset64, pSymbol)) {
+        DWORD dwOffset = 0;
+        IMAGEHLP_LINE64 line;
+        line.SizeOfStruct = sizeof(IMAGEHLP_LINE64);
```

Comment:
Either sizeof(<type>) or sizeof(<variable>) work OK I believe, the MSDN examples uses sizeof(<type>). Is sizeof(<variable>) preferred?